engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -30,13 +30,16 @@ public RenderGraph() {
         nodes = Lists.newArrayList();
     }
 
-    public String addNode(Node node, String suggestedId) { // TODO: Change suggestedId to SimpleUri
+    public SimpleUri addNode(Node node, SimpleUri suggestedId) {
```

Comment:
Hmmm... actually I'm thinking that in this case a string for `suggestedId` is more appropriate: the namespace eventually will be assigned automatically and shouldn't be provided here - for the time being `addNode` can add the `engine:` namespace, until we begin work on the API proper. 

But it's good that the method returns a SimpleUri.